tokio/tests/macros_test.rs

```diff
@@ -42,6 +42,7 @@ fn trait_method() {
 
 // https://github.com/tokio-rs/tokio/issues/4175
 #[tokio::main]
+#[allow(unreachable_code)]
```

Comment:
I don't think this lint should be triggered here (since [the same function without tokio::main doesn't receive unreachable_code](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=71057ad4a71364a7011dd9d69242a4c9)).

Reply:
Hmm... That's interesting. For some reason, it didn't trigger in my test environment.